### ISSUE TITLE
Empty transform plugin leads to false-positive missing-function-docstring issues

### ISSUE THREAD

**Issue by efroemling** [pylint-dev/pylint#10352]
### Bug description

```python
I have recently started replacing typing.TypeVar/typing.Generic usage with Python 3.12 style [type-params](https://docs.python.org/3/whatsnew/3.12.html#whatsnew312-pep695) in my codebase, and it has led to some odd new false positive issues being detected by pylint.

I have distilled things down to as simple a repro case as I can. It is worth noting that I am using a custom pylint transform plugin in my codebase and I found the error only occurs with it enabled. Normally I would assume my plugin is at fault, however I also found that the error persists even if I strip the plugin down to do nothing at all, so I figured it would still be worth submitting.
```

### Configuration

```ini
To repro the issue, place the following code in 'pylintbug.py':


"""Example of slight wonkiness with type params."""
# pylint: disable=too-few-public-methods

from typing import TypeVar, Generic


# ------------------------------------------------------------------------------
# Using new-style (Python 3.12) template parameters results in a
# complaint about missing docstrings in a method we override.
#
# This complaint seems very dependent on these specific conditions. If
# we disable the empty pylint plugin, take the middle class out of the
# equation, remove the __init__() from the child class, or even switch
# the order of the do_something_* overrides in the child class, the
# complaint goes away.

class BaseClass[T]:
    """A generic class."""

    def do_something(self) -> None:
        """Method meant to be overridden by child classes."""

class MiddleClass[T](BaseClass[T]):
    """Inherits from base generic class."""

    def do_something_2(self) -> None:
        """Method meant to be overridden by child classes."""


class ChildClass(MiddleClass[int]):
    """Concrete subclass of the middle one above."""

    def __init__(self) -> None:
        pass

    # Trying to override base class method results in pylint complaining
    # about a missing docstring.
    def do_something(self) -> None:
        pass

    # Trying to override middle class method does not complain about
    # missing docstring.
    def do_something_2(self) -> None:
        pass


# ------------------------------------------------------------------------------
# If we define the equivalent setup using old style TypeVar/Generic, we
# get no missing-docstring complaints.

T = TypeVar('T')

class BaseClassOld(Generic[T]):
    """A generic class using old-style type-vars."""

    def do_something(self) -> None:
        """Method meant to be overridden by child classes."""

class MiddleClassOld(BaseClassOld[T]):
    """Inherits from base generic class."""

    def do_something_2(self) -> None:
        """Method meant to be overridden by child classes."""


class ChildClassOld(MiddleClassOld[int]):
    """Concrete subclass of the middle one above."""

    def __init__(self) -> None:
        pass

    # In this case, neither method override results in missing-docstring
    # complaints.

    def do_something(self) -> None:
        pass

    def do_something_2(self) -> None:
        pass


Also place the following code in 'pylintbugplug.py':

"""A transform plugin that does nothing.

Strangely the issue only shows up when this plugin is active.
"""

from typing import Any

import astroid
import astroid.nodes
import astroid.manager


def func_annotations_filter(node: astroid.nodes.NodeNG) -> astroid.nodes.NodeNG:
    """Filter FunctionDefs"""

    # Do nothing.
    return node


def register(linter: Any) -> None:  # pylint: disable=unused-argument
    """Unused here; we're only registering plugins."""


def register_plugins(manager: astroid.manager.AstroidManager) -> None:
    """Register a transform for FunctionDef that does nothing."""
    manager.register_transform(astroid.FunctionDef, func_annotations_filter)


register_plugins(astroid.MANAGER)
```

### Command used

```shell
Now run the following command from directory containing pylintbugplug.py and pylintbug.py:
'''PYTHONPATH=. pylint --load-plugins=pylintbugplug pylintbug.py'''
```

### Pylint output

```python
************* Module pylintbug
pylintbug.py:38:4: C0116: Missing function or method docstring (missing-function-docstring)
```

### Expected behavior

I would expect the test case to not complain about any missing docstrings since all methods without docstrings are overrides.
You can see in the example code that the old style setup using typing.Generic/typing.TypeVar behaves as expected, but the new style type-param equivalent does not.

### Pylint version

```shell
pylint 3.3.6
astroid 3.3.9
Python 3.12.10 (main, Apr  8 2025, 11:35:47) [Clang 16.0.0 (clang-1600.0.26.6)]
```

### OS / Environment

Sequoia 15.4.1 on Apple Silicon 

### Additional dependencies

```python

```
